### PR TITLE
CI: Fix 01-test-utilities.t

### PR DESCRIPTION
**Pull request by perlpunk**
It can happen that the warning in the subprocess happens before the
SIGCHLD handler was installed.

The sleep will fix this in most cases.

(To be really sure one would have to add some method in Utils.pm to check if
the handler has been installed.)

Also I changed combined_like to combined_from, which made it easier to debug
the actual output.

Issue: https://progress.opensuse.org/issues/92164

I ran this in a loop in CircleCI [50 times](https://app.circleci.com/pipelines/github/perlpunk/openQA/171/workflows/a48cf629-ecf2-4ec4-9a1c-f238f0b26968/jobs/751), it always passed.

### CODE REVIEW

**Review by Martchus — APPROVED** [review]

**Comment by Martchus** [comment]
> (To be really sure one would have to add some method in Utils.pm to check if
the handler has been installed.)

And likely that wouldn't be that simple. I mean, what do we do if the check determines that the handler hasn't been installed but the child process has already terminated? This is still a situation we can't really handle keeping a positive test result.

We could register the handler before starting the child process (without knowing the PID and recording unknown PIDs) and when the child PID is known we make the handler aware of it but also handle the case when the handler was already called. That's of course rather complicated and maybe not worth it.

File: t/01-test-utilities.t [hunk 1]
```diff
@@ -35,16 +35,18 @@ subtest 'warnings in sub processes are fatal test failures' => sub {
             isnt(shift, 0, 'exit code of test failure is non-zero');
             $test_would_have_failed = 1;
         });
-    combined_like {
+    my $out = combined_from {
         # start a sub process like the test helper do and simulate a Perl warning
         OpenQA::Test::Utils::_setup_sigchld_handler 'test-process-1', start sub {
+            sleep 1;                                                     # Give Utils.pm a chance to install $SIG{CHLD}
```
Comment (okurz) [inline t/01-test-utilities.t:41]: We now need an "uncoverable statement" in the comment as well

**Comment by okurz** [comment]
Meh, those god-forsaken sleep calls. Does this slow down tests by second?

File: t/01-test-utilities.t (hunk 1 above)
Comment (perlpunk) [reply t/01-test-utilities.t]: ah right. done

**Review by okurz — APPROVED** [review]

**Pull request merged**